cookbook/case_studies/bioinformatics/blast/blastx_example.py

```diff
@@ -2,31 +2,50 @@
 BLASTX Example
 --------------
 
-This example will use BLASTX to search for a nucleotide sequence against a local protein database.
+This demonstration will utilize BLASTX to search for a nucleotide sequence within a local protein database.
 """
 
 # %%
-# First, we need to import some libraries.
+# Import the necessary libraries.
+from pathlib import Path
 from typing import NamedTuple
 
 import matplotlib.pyplot as plt
 import pandas as pd
+import requests
 from flytekit import conditional, kwtypes, task, workflow
 from flytekit.extras.tasks.shell import OutputLocation, ShellTask
 from flytekit.types.file import FlyteFile, PNGImageFile
 
+
 # %%
-# A ``ShellTask`` is useful to run commands on the shell.
-# In this example, we use ``ShellTask`` to generate and run the BLASTX command.
+# Download the data from GitHub.
 #
-# First, we define the location of the BLAST output file.
-# Then we define variables that contain paths to: the input query sequence file, the database we are searching against, and the file containing the BLAST output.
-# Finally, we generate and run the BLASTX command. Both ``stdout`` and ``stderr`` are captured and saved to the ``stdout`` variable.
+# .. note::
+#   When running the code on the demo cluster, make sure the data is included in the Docker image.
+def download_dataset():
+    Path("kitasatospora").mkdir(exist_ok=True)
```

Comment:
looks like CI needs access to this dataset: https://github.com/flyteorg/flytesnacks/actions/runs/4123721438/jobs/7122103442#step:4:23031

```
COPY failed: file not found in build context or excluded by .dockerignore: stat blast/kitasatospora: file does not exist
```